IPython/terminal/embed.py

```diff
@@ -83,6 +96,8 @@ def __init__(self, **kw):
 
         super(InteractiveShellEmbed,self).__init__(**kw)
 
+        frame = sys._getframe(InteractiveShellEmbed._stack_frames)
+        self._call_location_id = '%s:%s' % (frame.f_code.co_filename, frame.f_lineno)
```

Comment:
I don't think this stack depth will be right if the user is instantiating `InteractiveShellEmbed` themselves - I assume you've calibrated it for calling `embed()`? Doing anything with stack depth is super brittle.